### ISSUE TITLE
UDD which of the two do you think is a cm ?

### ISSUE THREAD

**Issue by timotree3** [hanabi/hanabi.github.io#651]
I'm copying https://github.com/Zamiell/hanabi-live/issues/2175

Original post:

@mmelwen

> ![image](https://user-images.githubusercontent.com/66402571/115789779-58014200-a3ce-11eb-988e-9da685803636.png)
> 
> https://hanab.live/replay/509002#1
> 
> First turn of the game I give yellow to Lel and so now he knows that are both the same card. Which card should Lel discard?
> There are two contradicting thoughts,
> 
> 1) should discard the focus of the clue, like UTD
> 2) should discard the left because is how we discard trash
> 
> I believe that this should be defined and the other one should be a cm.

@Lel0uch-H

> In addition to this, if the focus of a UDD is on chop, touches exactly 2 cards and they are both known to be dupes to the clue receiver but not globally, the problem becomes even more complex.
> 
> Looking for more opinions on choosing how to define the discard order in such cases.

@timotree3 

> 
> Obviously the expected discard order should be discarding the card that is guaranteed trash by a blind play (the focus of the clue) before any other cards. This also applies to trash bluffs, and is [included in the doc](https://hanabi.github.io/docs/level-13/#the-trash-bluff):
> 
> > In the case where multiple cards are clued as part of a Trash Bluff, all of the touched cards are considered to be trash. However, the focus of the clue should be the first thing discarded, which allows for the possibility of your teammates to give a follow-up Fix Clue.
> 
> If the card besides the focus of the clue is discarded, then that should be a trash order chop move, because just communicating that the clue receiver know it's not a UTD because they see all the copies of those cards is not valuable.
> 
> Here are two in-game examples where blindly obeying left to right discard order leads to confusion/missed opportunities with chop moves:
> 
> 1. https://github.com/hanabi/hanabi.github.io/issues/419#issuecomment-823348262
> 2. https://hanab.live/replay/482822#11
>    timotree has two clued 1's that were clued as a trash chop move. They are known trash because kinkajusrevenge has a copy of m1 clued. If the focus of the clue is rainbow 1, then kinkaju thinks it's a play clue, not a trash order chop move, therefore it's important to discard the focus of the clue (slot 3) first to keep kinkaju in sync. timo discards slot 3, and half of the players write [cm] on orucsoraihc's chop, believing that timo was expected to discard slot 2.

**Comment by timotree3** [comment]
> If they're globally promised as dupes then I'd much prefer to allow TOCMs with the pair.
> ...
> So if the difference between these conventions are error-prone, then I guess we should prescribe always discarding the focus first :/

Are you thinking that discarding the focus first prevents TOCMs with the pair? It doesn't prevent TOCMs; you just discard the non-focus in order to TOCM.

**Comment by pianoblook** [comment]
> Are you thinking that discarding the focus first prevents TOCMs with the pair? It doesn't prevent TOCMs; you just discard the non-focus in order to TOCM.

If the focus was on chop and I discard my left-most, I would be very hesitant to safely assume that it would work as a OCM. In other words I'm skeptical that we should allow TOCMs from UDD if we don't also maintain the (much more familiar) left-discard assumption

**Comment by pianoblook** [comment]
Also based on recent experiences of players assuming UDD even in situations where it could have just been a valid UTD, I'd also be very skeptical of encouraging people to try and perform TOCMs after a Discharge.

**Comment by timotree3** [comment]
> If the focus was on chop and I discard my left-most, I would be very hesitant to safely assume that it would work as a OCM. In other words I'm skeptical that we should allow TOCMs from UDD if we don't also maintain the (much more familiar) left-discard assumption

I don't get your hesitancy or skepticism. We can just define the expected discard order to be focus first then leftmost and then discarding the leftmost in that scenario is clearly a TOCM. 

I think your experience with players misreading UTD's suggests "focus first then leftmost" over anything else: If a chop focus UDD happens that can't possibly be a UTD, but the player thinks it's a UTD and discards the focus first, then we have a TOCM in the leftmost first convention and nothing in focus first.

**Comment by pianoblook** [comment]
Let me see if I can put my thoughts another way:

In a world with zero risk of mistakes, there's no reason why a known UDD shouldn't result in both the cards being equally weighted as trash. And as long as it *could* be just a UTD, no TOCMs could ever occur anyway.

Therefore - as I understand it at least - the only value of making a 'Focus First' rule (instead of 'left-most') is to help mitigate the risk of the clued player forgetting that it could have just been a UTD and 'naturally' discarding the left-most.

*But* my worry is that this would open up a *different*, and IMO probably more common, mistake: now suddenly the expected order of these discards will often be opposite of how all other TOCMs work in the game. Call this a _Frequency Bias_ issue or w/e.

 I'd expect both of the following scenarios to happen a dangerous amount:
> A) Alice discards her right-most of the 2 cards (it was the focus). Say from her perspective it might have been a UTD and she therefore had no option. Bob, though, sees that it was indeed a UDD and naturally assumes this was therefore a TOCM. [this happened in game recently]
> B) Alice *knows* it was a UDD, so she naturally discards the right-most one instead of the left in an attempt to TOCM Bob. Bob, though, happened to read this github page and knows that this shouldn't be a TOCM. He discards a 5.

------
 So basically it just comes down to the following question, imo:

What's more likely, and therefore which do we want to design towards preventing:
[A] that mistakes will happen due to players assuming UDD when UTD was a possibility
=OR=
[B] that mistakes will happen due to forgetting that UDD order can run opposite of other TOCMs

**Comment by Zamiell** [comment]
> There are two contradicting thoughts,

@mmelwen  @Lel0uch-H this is already defined.
see [the last bullet point of the first example in the UDD section](https://hanabi.github.io/docs/extras/discharges/#unknown-dupe-discharge-udd).
i.e. players should discard the focus of the clue

as timo also points out, this is **also** defined for Trash Bluffs in the same same way.

thus, TOCMs in this way have a good chance of working as-is for level-23 players who are paying close attention.

so the issue at hand is whether or not to make a dedicated section for specific type of TOCM.

pianoblook is raising fantastic points, namely that having TOCM turned on in this situation allows some slack for the team to not become desynchronized in the event of a minor-error

**Comment by Zamiell** [comment]
such a section might read something like this:

### No *Trash Order Chop Moves* After *Trash Finesses* / *Trash Bluffs* / *Unknown Dupe Discharges*

- Sometimes, a *Trash Finesse* or *Trash Bluff* or *Unknown Dupe Discharge* will touch multiple cards. When this happens, the player who received the clue is supposed to discard the focus of the clue first.
- Furthermore, all of these moves promise the clued player that all of the touched cards are safe to discard. Thus, it would make sense that the clued player could perform a *Trash Order Chop Move* by choosing to discard a non-focused card first.
- However, the group agrees that players should not ever *Chop Move* in this situation, because this kind of thing is confusing and someone might be making a small mistake.
- For example, in a 3-player game:
  - All the 1's are played except for the red 1.
  - Cathy's hand is completely unclued.
  - Alice clues number 1 to Cathy, touching a *Trash* 1 on slot 1 and a *Trash* 1 on slot 5. This clue is *Chop-Focused* on Cathy's slot 5.
  - Bob sees that if he does nothing, Cathy will go on to misplay the 1 on her slot 5 as a red 1.
  - Thus, Bob knows that he must have the red 1 as a *Trash Finesse*. Bob plays his *Finesse Position*. It is a red 1 and it successfully plays on the stacks.
  - Cathy now knows that Alice performed a *Trash Finesse*. Furthermore, Cathy has nothing to do and needs to discard. Cathy knows that she is supposed to discard the trash 1 that was the focus of Alice's clue.
  - However, Cathy has a problem: she is unsure what slot Alice's clue was focused on. Normally, Alice's clue would be *Chop-Focused* on Cathy's slot 5. But if Bob did not do anything, then Cathy would have probably played her slot 1 card, interpreting the clue as a *Stale 1's Clue*. If the 1 on slot 5 was *Stale*, then the real focus of Alice's clue would be on slot 1.
  - Cathy assumes that the clue was *Stale* and discards slot 1.
  - Alice intended for her clue to be focused on slot 5. So when Alice sees that Cathy discarded the non-focused slot, she is tempted to think that Cathy is performing a *Trash Order Chop Move*.
  - However, Alice knows that in this situation, players are explicitly not supposed to *Chop Move* anything. It is possible that Cathy forgot that she was supposed to discard the focus of the clue first and was just discarding from left-to-right like normal. Or, it is possible that Cathy messed up the focus of the clue by incorrectly applying *Focus Inversion*. Or, it is possible that Cathy made some other small mistake.

**Comment by Zamiell** [comment]
one problem with this is that it could be unclear when TOCM should apply and when it should not apply. not sure what others think.

another option here is to (continue) to leave this undefined and not edit the doc at all.

**Comment by timotree3** [comment]
The stale 1's scenario is interesting but I think that if we created a convention that defines the expected discard as the rigid focus of the clue ignoring focus inversion, then that situation will not be error prone. It's perfectly fine to define the expected discard as the rigid focus of the clue because if it is in fact a stale card, then it's trash anyway. (except in pink, but we really shouldn't do trash bluffs that focus useful pink cards)

Disabling TOCM for trash bluffs, finesses, and UDDs seems quite extreme, especially given that such TOCMs play quite a big role in preventing bottom deck losses in my estimation.

Also, creating a convention that says "ignore signals from your teammates under certain circumstances" can create more harm than good because some players will choose not to ignore the signal while others will ignore it, creating just as much desync as the convention tries to prevent.

I don't think it's very difficult to discard the focus of a clue first. Speaking for myself, I am so used to determining the focus of a clue and remembering it for later (e.g. for pink promise) that it is second nature to remember which clued cards have been focused. I assume it is habitual for most other players as well.

**Comment by Zamiell** [comment]
> if we created a convention that defines the expected discard as the rigid focus of the clue ignoring focus inversion, then that situation will not be error prone

that's interesting.
i agree it would be less-error prone.
but it would still be error-prone in the situation where cathy forgets to discard the focus of the clue first and just does the normal/habitual thing of discarding from left-to-right without thinking too hard.
(i could adjust the above example accordingly.)

i mean, melwen/lelouch above didn't even know the focus convention existed, so that is pretty good proof that the it is error-prone =p

> Disabling TOCM for trash bluffs, finesses, and UDDs seems quite extreme

to clarify, the section i wrote would only apply to trash bluffs, finesses, and UDDs where *the focus of the clue is unclear*.
so, it is a subset of what you wrote.
more specifically: it would be relegated to situations where the clue touches chop and also touches one other card.

> creating a convention that says "ignore signals from your teammates under certain circumstances" can create more harm than good because some players will choose not to ignore the signal while others will ignore it

i agree with you and don't really like it either.
perhaps a better solution is to simplify, simplify, simplify: make discarding for all 3 of these conventions always ignore focus.
the pro is that you reduce TOCM confusion, the con is that you make *Trash Finesses* and co. less versatile

**Comment by Zamiell** [comment]
for the record, this reconfiguration would be independent of whether to allow UDD-style TOCM.
piano's point about accidental good-card-discarding would still apply whether or not we reconfigure the aforementioned 3 moves.

**Comment by Zamiell** [comment]
oh, you actually couldn't reconfigure UDD in this way, you could only reconfigure Trash Finesse & Trash Bluff.
this is because for UDD specifically, in the case of touching 3 cards with 1 card on chop, then you **must** discard the trash one, otherwise you could be discarding a critical card.

that would mean that after such a reconfiguration, UDD would not have parity with Trash Finesse & Trash Bluff.
maybe that's a bit confusing or maybe it doesn't matter?

**Comment by pianoblook** [comment]
> but it would still be error-prone in the situation where cathy forgets to discard the focus of the clue first and **just does the normal/habitual thing of discarding from left-to-right** without thinking too hard.

I maintain that this error would still be more common, and costly. We almost never give trash finesses in which the secondary 1 is actually playable (with the possible exception, imo, of Pink variants). Frankly I find it hard to imagine when that would ever be a good idea outside of weird emergencies. In fact we can delete the almost completely w/ the conventions from the new #653 proposal.

With the conventions as is, I'd still be much more inclined to TOCM as Bob if my Alice discarded her right-most of two 1s after a trash finesse

**Comment by Zamiell** [comment]
> I'd still be much more inclined to TOCM as Bob if my Alice discarded her right-most of two 1s after a trash finesse

to clarify, i dont think anyone would be confused in the situation of:
- 3 player game
- Cathy has a completely unclued hand
- Alice clues 1's to Cathy, touching slot 1 and slot 2
- Bob blind-plays red 1
- Cathy discards slot 2 as a TOCM

that would work as a TOCM in the current conventions, and would certainly continue to work regardless of whatever else is decided in this thread

**Comment by timotree3** [comment]
> I maintain that this error would still be more common, and costly. We almost never give trash finesses in which the secondary 1 is actually playable (with the possible exception, imo, of Pink variants). Frankly I find it hard to imagine when that would ever be a good idea outside of weird emergencies. In fact we can delete them almost completely w/ the conventions from the new #653 proposal.

I don't want to discard the focus of the clue first because the other cards might not be trash, although it's true. I want to discard the focus of the clue first because it is the most likely card to be known trash (i.e. when there is ambiguity between UDD and UTD) and because in some cases it is necessary to demonstrate to the team that the clue was not a normal play clue (as in reverse trash finesses and the second example I linked above)

**Comment by pianoblook** [comment]
Revised statement then: 

*I'd still be much more inclined to TOCM as Bob if my Alice discarded her right-most of two 1s after a trash finesse, **even if the right-most was technically the focus.**

This is because from my experience the ability to TOCM people comes up extremely often, and in the vast majority of these situations this is performed by discarding the right-most of the trash. 

Even though in the scenario in which the focus is also the right-most, and hence there might exist a 1-in-1000 chance that their left-most 1 is actually useful, it feels a lot more intuitive - and thus more likely - to discard the leftmost one if you're not trying to perform a TOCM

**Comment by Zamiell** [comment]
piano, what is your reaction to the new asymmetry between UDD and TF/TB in such a reconfiguration

**Comment by timotree3** [comment]
piano and I had some discussion in DMs and he agreed that these are the pros and cons of the 3 options currently presented:

focus first, then leftmost:
- pro: allows TOCMs on asymmetrically known UDDs in a way that has no room for missing some asymmetric information when applying it (+1 free chop move ~1/100 games)
- pro: ensures that trash finesses always resolve correctly (prevents bomb/desync ~1/1000 games)
- pro: ensures that ambiguous play clues/trash clues always resolve correctly in a way that has no room for missing some asymmetric information when applying it (prevents bomb/desync ~1/1000 games)
- pro: ensures that trash bluff lies have the most time to be fixed (prevents discarding a playable ~1/500 games)
- con: people might forget the convention, causing desync (rightmost focus trash clues happen ~1/20 games)

leftmost first:
- pro: we're already used to it, so we won't forget the convention
- con: does not allow TOCMs on chop-focus asymmetrically known UDDs. (-1 chop move ~1/100 games)
- con: players may attempt TOCMs on chop-focus asymmetrically known UDDs, causing desync (causes desync ~1/100 games)

leftmost first + no TOCMs on UDD cards
- pro: players are told not to attempt TOCMs on chop-focus asymmetrically known UDDs, which would otherwise cause desync in leftmost first but not focus first (prevents desync ~1/100 games)
- con: does not allow TOCMs on any UDD, even the ones that would otherwise be fine (-1 free chop move ~1/3 games)
- con: players may attempt a TOCM anyway, causing desync (UDD's happen ~1/3 games)

My conclusion is that while focus first is the optimal convention, it will only produce better results than the status quo for a general audience if players will begin remembering the convention such that it fails in less than 1/5 of the cases where it applies.

Not sure if that's achievable. Players do seem to be able to reliably remember edge case conventions that come up often enough though (e.g. 1's play order chop-focus).

Is it possible to experiment with this? Try implementing it and see if people start remembering it after a couple weeks of playing with it?

**Comment by Zamiell** [comment]
> con: players may attempt a TOCM anyway, causing desync (UDD's happen ~1/3 games)

while UDDs are around 1/3 games or 1/4 games, UDD TOCMs are a small subset of that
so wouldn't it be more accurate to state this as ~1/12 or ~1/16 ?

> Is it possible to experiment with this?

you are free to experiment with any of the proposed conventions that you want, just get your teammates to agree before the game starts

**Comment by sjdrodge** [comment]
This one seems like a no-brainer to me. Focus first trash discard should just become the norm for these moves and TOCM happens when you discard differently. If we can expect players to remember that the 1's play order is Chop -> LtR fresh -> RtL starting hand, then we should expect this convention to work too.

**Comment by Zamiell** [comment]
tim, did you experiment?

**Comment by Zamiell** [comment]
doc changes are as follows:

trash bluff - already says to discard focus

trash finesse - added this:
```
- In the case where multiple cards are clued as part of a *Trash Finesse*, **all** of the touched cards are considered to be trash. However, the focus of the clue should be discarded first. (This is the exact same policy as what happens in a *Trash Bluff*.)
```

UDD - added this:
```
- Remember that after an *Unknown Dupe Disharge*, the player who received the clue is supposed to *discard the focus of the clue*.
  - However, in the case where only two brand-new cards are touched, then the player who received the clue knows that both of these cards must be the same. In this special case, they can discard the non-focused card to cause a *Trash Order Chop Move*.
```

UTD - nothing needs to be added because only the focus is promised to be trash

is that everything?
